server/base/src/main/java/org/apache/accumulo/server/ServerInfo.java

```diff
@@ -0,0 +1,285 @@
+    log.info("Instance " + instanceID);
+    try {
+      Accumulo.init(volumeManager, getInstanceID(), getServerConfFactory(), applicationName);
+    } catch (IOException e) {
+      throw new IllegalStateException(e);
+    }
+    MetricsSystemHelper.configure(applicationClassName);
+    DistributedTrace.enable(hostname, applicationName,
+        getServerConfFactory().getSystemConfiguration());
+  }
+
+  public VolumeManager getVolumeManager() {
+    return volumeManager;
+  }
+
+  public String getInstanceID() {
+    return instanceID;
+  }
+
+  public String getZooKeeperRoot() {
+    return zooKeeperRoot;
+  }
+
+  public String getZooKeepers() {
+    return zooKeepers;
+  }
+
+  public int getZooKeepersSessionTimeOut() {
+    return zooKeepersSessionTimeOut;
+  }
+
+  @Override
+  public String getPrincipal() {
+    return getCredentials().getPrincipal();
+  }
+
+  @Override
+  public AuthenticationToken getAuthenticationToken() {
+    return getCredentials().getToken();
+  }
+
+  @Override
+  public boolean saslEnabled() {
+    return getServerConfFactory().getSystemConfiguration()
+        .getBoolean(Property.INSTANCE_RPC_SASL_ENABLED);
+  }
+
+  @Override
+  public Properties getProperties() {
+    Properties properties = new Properties();
+    properties.setProperty(ClientProperty.INSTANCE_ZOOKEEPERS.getKey(), getZooKeepers());
+    properties.setProperty(ClientProperty.INSTANCE_ZOOKEEPERS_TIMEOUT.getKey(),
+        Integer.toString(getZooKeepersSessionTimeOut()));
+    properties.setProperty(ClientProperty.INSTANCE_NAME.getKey(), getInstanceName());
+    ClientProperty.setAuthenticationToken(properties, getAuthenticationToken());
+    properties.setProperty(ClientProperty.AUTH_PRINCIPAL.getKey(), getPrincipal());
+    return properties;
+  }
+
+  public String getInstanceName() {
+    return instanceName;
+  }
+
+  public Credentials getCredentials() {
+    return SystemCredentials.get(getInstanceID());
+  }
+
+  public Connector getConnector(String principal, AuthenticationToken token)
+      throws AccumuloSecurityException, AccumuloException {
+    return Connector.builder().forInstance(instanceName, zooKeepers).usingToken(principal, token)
+        .withZkTimeout(zooKeepersSessionTimeOut).build();
+  }
+
+  public synchronized ServerConfigurationFactory getServerConfFactory() {
+    if (serverConfFactory == null) {
+      serverConfFactory = new ServerConfigurationFactory(this);
+    }
+    return serverConfFactory;
+  }
+
+  public String getApplicationName() {
+    Objects.requireNonNull(applicationName);
+    return applicationName;
+  }
+
+  public String getApplicationClassName() {
+    Objects.requireNonNull(applicationClassName);
+    return applicationName;
+  }
+
+  public String getHostname() {
+    Objects.requireNonNull(hostname);
+    return hostname;
+  }
+
+  public void teardownServer() {
+    DistributedTrace.disable();
+  }
+
+  public List<String> getMasterLocations() {
+
```

Comment:
Is this code a duplicate of what is in ClientContext?  If so, could this be consolidated?

Reply:
Fixed. Only exists in ClientContext now.